lib/curl_desc_pool.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#ifndef SNOWFLAKE_CURL_DESC_POOL_H
+#define SNOWFLAKE_CURL_DESC_POOL_H
+
+#include <curl/curl.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    /**
+     * Get curl desc instance from pool.
+     *
+     * @param url       The url of the rest request
+     * @param proxy     The proxy setting, null if not available.
+     * @param no_proxy  The proxy setting, null if not available.
+     *
+     * @return curl desc instance from pool
+     */
+    void* get_curl_desc_from_pool(const char* url, const char* proxy, const char* no_proxy);
```

Comment:
can we have type for curl_desc also here instead of void?

Reply:
This one actually no. 
This is C interface wrapping C++ implementation and the return value is point to C++ instance defined here:
https://github.com/snowflakedb/libsnowflakeclient/pull/718/files#diff-2e87c957306939dd5314f808ebfef961e5b2468f67ca6f2dd6cba9b5d2ae50e3R15
```
  struct CURL_DESC_S {
    CurlDescPool::SubPool* sub_pool;
    std::unique_ptr<CurlDesc> curl_desc;
  };
```
To reuse the C++ implementation in both libsnowflakeclient(C API) and odbc (C++) this is something we can't avoid.